### ISSUE TITLE
Map iteration allocates closures

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/hashlink#64]
According to profiling data, iterating on Map allocates many closures (to fix).

```haxe

class Main  {

	
	static function main() {
		var m = [for( i in 0...10000 ) ""+i => i];
		hl.Profile.enable = true;
		var k = 0;
		for( v in m )
			k += v;
		hl.Profile.enable = false;
		trace( hl.Profile.getData() );
	}

}
```

**Comment by ncannasse** [comment]
Closed with e2e731b6c2a1ee63ec6dc502bc4e489fe7e6dacc

**Comment by ncannasse** [comment]
https://github.com/HaxeFoundation/haxe/commit/e2e731b6c2a1ee63ec6dc502bc4e489fe7e6dacc
